File: python/code_examples/01-variables-and-types/main.js
// JavaScript: let/const (modern); undefined = uninitialized, null = intentional absence
const name = "Ada";
let count = 42;
const pi = 3.14;
let missing = null; // intentional "no object"
let unset; // undefined

function main() {
  console.log(name, count, pi, missing, unset);
}

main();

File: python/code_examples/01-variables-and-types/main.py
# Variables: Python uses dynamic typing; None is the single "no value" sentinel.
name = "Ada"
count = 42
pi = 3.14
missing = None  # like null in other languages (only one None object)

# Type hints (optional, for tooling — not enforced at runtime by default)
age: int = 30


def main() -> None:
    print(name, count, pi, missing, age)


if __name__ == "__main__":
    main()
